### ISSUE TITLE
Option to omit redundant size directives

### ISSUE THREAD

**Issue by nitrocaster** [x64dbg/x64dbg#1342]
Consider the following example:

`mov eax, dword ptr ds:[ecx]`

Since the destination is `eax`, which is 32-bit register, and the instruction is plain `mov`, it's obvious that `[ecx]` is a 32-bit value. IDA Pro omits such redundant directives and it would be nice to have this option in x64dbg.

**Comment by mrexodia** [comment]
You would think it's obvious. Try writing an assembler and you will know
otherwise. Ask @Nukem9

**Comment by nitrocaster** [comment]
I reverse engineer stuff and occasionally write in x86 assembly. In this particular case it is obviously redundant and makes the code too verbose, not helping to read and understand it. I believe that is the reason why IDA hides it.
Anyway, I suggest to add an *option*, not just omit these directives. Please note that I suggest omit only redundant ones, like the one in the first post. For example, `mov dword ptr ds[ecx], 5` size directive is absolutely necessary because `5` could be anything from byte to dword.

**Comment by mrexodia** [comment]
Yes my point is that deciding which directives are redundant is a hard
problem. I could of course add an option to hide the whole size directive
very easily. It doesn't matter much for your understanding anyway.

A much better solution in my opinion is a plugin that shows instructions as
pseudo code next to the correct behavior to aid you in that way.

**Comment by nitrocaster** [comment]
> It doesn't matter much for your understanding anyway.

It matters in one case but not in the other. I provided an example for each case. Hope you understand now.

**Comment by mrexodia** [comment]
I already understood the issue after the first post, but I wouldn't know a good heuristic to determine when a size specifier is redundant. The most trivial case `mov reg32, dword ptr ds:[mem]` might work, but I'm not sure if overloaded variations exist where equal memory sizes are important or not the general case. The solution assembler-wise is still a pure hack and doesn't work in all cases (look for issues with 'XEDParse' if you're interested).
